operator/config/crd/patches/webhook_in_runtimetasks.yaml

```diff
@@ -0,0 +1,17 @@
+# The following patch enables conversion webhook for CRD
+# CRD conversion requires k8s 1.13 or later.
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: runtimetasks.operator.kubeadm.x-k8s.io
+spec:
+  conversion:
+    strategy: Webhook
+    webhookClientConfig:
+      # this is "\n" used as a placeholder, otherwise it will be rejected by the apiserver for being blank,
+      # but we're going to set it later using the cert-manager (or potentially a patch if not using cert-manager)
+      caBundle: Cg==
+      service:
+        namespace: system
+        name: webhook-service
+        path: /convert
```

Comment:
what is our upgrade / conversion story for user API objects using conversion webhooks?
i.e. we need to support proper conversion.

AFAIK, CAPI did not support that very well between v1alpha1 -> 2

Reply:
v1alpha1->2 was a disruptive change because e.g. the API group changed

For "normal" changes, you can leverage on conversion webhooks https://book.kubebuilder.io/multiversion-tutorial/conversion.html (same approach CAPI will use for v1alpha2 -> v1alpha3)

Reply:
> For "normal" changes, you can leverage on conversion webhooks https://book.kubebuilder.io/multiversion-tutorial/conversion.html (same approach CAPI will use for v1alpha2 -> v1alpha3)

ack